packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMainInput.tsx

```diff
@@ -96,14 +99,22 @@ function SourceChainTokenBalance({
     ? selectedTokenBalances.parentBalance
     : selectedTokenBalances.childBalance
 
+  const selectedTokenDecimals = useMemo(() => {
+    if (selectedToken) {
+      return selectedToken.decimals
+    }
+
+    return nativeCurrencyDecimalsOnSourceChain
+  }, [nativeCurrencyDecimalsOnSourceChain, selectedToken])
```

Comment:
we have a hook called `useSelectedTokenDecimals()`, so you should update it with this code and use it

Reply:
done